src/test/java/org/apache/cassandra/sidecar/restore/RestoreJobManagerTest.java

```diff
@@ -66,7 +68,7 @@ class RestoreJobManagerTest
     @BeforeEach
     void setup()
     {
-        Injector injector = Guice.createInjector(new MainModule());
+        Injector injector = Guice.createInjector(Modules.override(new MainModule()).with(new TestModule()));
```

Comment:
why do we need the test module here? the metrics should be provided in the main module

Reply:
Here it is needed because, the test doesn't provide all the classes we need. It only provides `mock(RestoreJobConfiguration.class)`. But for building Vert.x instance we need sidecar configuration object. The test throws error saying, yaml not found if we use `MainModule`. `TestModule` provides  `SidecarConfiguration` hence we avoid yaml error.

Reply:
maybe we need to add these fields to the yaml?